network/p2p/pubsub_test.go

```diff
@@ -42,10 +43,13 @@ func (suite *PubSubTestSuite) SetupTest() {
 }
 
 type mockDiscovery struct {
-	peers []peer.AddrInfo
+	peerLock sync.Mutex
+	peers    []peer.AddrInfo
 }
 
 func (s *mockDiscovery) SetPeers(peers []peer.AddrInfo) {
+	defer s.peerLock.Unlock()
+	s.peerLock.Lock()
```

Comment:
nit: probably doesn't matter here, but in general I think it's best practice to reverse the order of these two lines: https://stackoverflow.com/a/51725166